cmd/client/main.go

```diff
@@ -149,7 +149,7 @@ func (c *Coordinator) doScrape(request *http.Request, client *http.Client) {
 
 	port := request.URL.Port()
 	if len(port) > 0 {
-		if *allowPort != "*" && *allowPort != port {
+		if *allowPort != "*" && !matchPort(*allowPort, port) {
```

Comment:
Edge case: what happens when `*` is provided as an argument (ex `--allow-port 8080,*,8081` or `--allow-port 8081 --allow-port '*'`).

I would argue that this should be an invalid specification and should result in an error.

Reply:
I would concur on it being an invalid state, so I would rather we error when `len(allowPorts) > 1 && strings.Contains(allowPorts, "*")`.